profiling/cmd/remote.go

```diff
@@ -0,0 +1,113 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+	ppb "google.golang.org/grpc/profiling/proto"
+)
+
+func setEnabled(ctx context.Context, c ppb.ProfilingClient, enabled bool) error {
+	_, err := c.Enable(ctx, &ppb.EnableRequest{Enabled: enabled})
+	if err != nil {
+		grpclog.Infof("error calling Enable: %v\n", err)
+		return err
+	}
+
+	grpclog.Infof("successfully set enabled = %v", enabled)
+	return nil
+}
+
+func retrieveSnapshot(ctx context.Context, c ppb.ProfilingClient, f string) error {
+	grpclog.Infof("establishing stream stats stream")
+	stream, err := c.GetStreamStats(ctx, &ppb.GetStreamStatsRequest{})
+	if err != nil {
+		grpclog.Errorf("error calling GetStreamStats: %v\n", err)
+		return err
+	}
+
+	s := &snapshot{StreamStats: make([]*ppb.Stat, 0)}
+
+	grpclog.Infof("receiving and processing stream stats")
+	for {
+		resp, err := stream.Recv()
```

Comment:
This is getting all stats between the previous call and this. It's actually a unary call, with a slice of stats in the response.

(If you want to make this a real streaming call, the server need to keep draining, and push the stats to the stream. Then either the server needs to monitor the timeout and stop sending, or the client needs to expect a RPC timeout)

Reply:
Yeah, I agree. I think @easwars pointed this out too (in the service PR).

I'll update this and the service PR to use a slice instead of a streaming RPC.

Reply:
Done.